lineapy/plugins/pipeline_writers.py

```diff
@@ -216,45 +223,106 @@ def _write_requirements(self) -> None:
         file.write_text(lib_names_text)
         logger.info(f"Generated requirements file: {file}")
 
-    def _write_module_test(self) -> None:
+    def _store_artval_for_testing(self) -> None:
+        """
+        (Re-)Store artifact values as pickle files to serve as "ground truths"
+        to compare against for equality evaluation of function outputs. The new
+        pickle files are to bear corresponding artifact names.
+        """
+        # Create subdirectory if nonexistent
+        dirpath = self.output_dir / self.test_artval_dirname
+        dirpath.mkdir(exist_ok=True, parents=True)
+
+        # Store each artifact value
+        for sa in self.artifact_collection.session_artifacts.values():
+            for art in sa.target_artifacts:
+                filepath = dirpath / f"{slugify(art.name)}.pkl"
+                artval = art.get_value()
+                with filepath.open("wb") as fp:
+                    pickle.dump(artval, fp)
+
+    def _write_module_test_scaffold(self) -> None:
         """
-        Write out test scaffolding for refactored code in module file.
-        The scaffolding contains placeholders for testing each function
+        Write out test scaffold for refactored code in module file.
+        The scaffold contains placeholders for testing each function
         in the module file and is meant to be fleshed out by the user
-        to suit their needs. When run out of the box, it simply tests
-        whether each function in the module runs without error.
+        to suit their needs. When run out of the box, it performs a naive
+        form of equality evaluation for each function's output,
+        which demands validation and customization by the user.
         """
-        # Format components to be passed into file template
-        module_name = f"{self.pipeline_name}_module"
-        test_class_name = f"Test{self.pipeline_name.title().replace('_', '')}"
+        # Extract information about each function in the pipeline module.
+        # This information is to be eventually passed into file template.
+        # Fields starting with an underscore are not intended for direct use
+        # in the template; they are meant to be used for deriving other fields.
+        # "output" here denotes artifact(s) or common variable(s) that
+        # the function generates.
```

Comment:
add example as discussed to go over the confusing key names in function_metadata_list

Reply:
Done